python/lsst/analysis/tools/actions/plot/elements/histElement.py

```diff
@@ -0,0 +1,77 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("HistElement",)
+
+from typing import TYPE_CHECKING
+
+from lsst.pex.config import Field
+
+from ....interfaces import PlotElement
+
+if TYPE_CHECKING:
+    from lsst.analysis.tools.interfaces import KeyedData
+    from matplotlib.axes import Axes
+
+
+class HistElement(PlotElement):
+    """Configuration options for ax.hist elements.
+
+    Attributes
+    ----------
+    valsKey : `~lsst.pex.config.Field`
+        Y-axis data vector key.
+    bins : `~lsst.pex.config.Field`
+        Number of x axis bins within plot x-range.
+    """
+
+    valsKey = Field[str](
+        doc="Y-axis data vector key.",
+        default="value",
+    )
+    bins = Field[int](
+        doc="Number of x axis bins within plot x-range.",
+        default=50,
+    )
+
+    def __call__(self, data: KeyedData, ax: Axes, **kwargs) -> KeyedData:
+        """Plot y versus x as lines and/or markers.
+
+        Parameters
+        ----------
+        data : `~lsst.analysis.tools.interfaces.KeyedData`
+            Keyed data containing the data to plot.
+        ax : `~matplotlib.axes.Axes`
+            Axes to plot on.
+
+        Returns
+        -------
+        data : `~lsst.analysis.tools.interfaces.KeyedData`
+            Data used for plotting.
+        """
+        ax.hist(
```

Comment:
It might be useful to have `cumulative` and `density` as options to pass to `ax.hist`.  Adding them now adds some extra utility to this element that might be useful in the future.

Reply:
Good idea. I've added `cumulative` and `density`, as well as `color` for good measure and to bring it into line with `scatterElement`.

I wasn't sure what type is appropriate for `cumulative`. I.e., a `bool` captures most use-cases, however, setting `cumulative` to `-1` returns an inverted cumulative histogram (i.e., starting at 1 and counting down). Is that a use-case worth considering? If so, we can change the type to `int` and add extra logic to capture that? Keen to hear your thoughts.

Reply:
I almost always just use `cumulative=True`, so that's enough for me on this ticket.